variables.tf

```diff
@@ -1,36 +1,164 @@
-#   - Use variable validation when possible
-#
+variable "cluster_id" {
+  type        = string
+  description = "The ID of the cluster you wish to deploy the agent in"
+}
+
+variable "cluster_resource_group_id" {
+  type        = string
+  description = "The Resource Group ID of the cluster"
+}
 
-variable "name" {
+variable "cluster_config_endpoint_type" {
+  description = "Specify which type of endpoint to use for for cluster config access: 'default', 'private', 'vpe', 'link'. 'default' value will use the default endpoint of the cluster."
   type        = string
-  description = "A descriptive name used to identify the resource instance."
+  default     = "default"
+  nullable    = false # use default if null is passed in
+  validation {
+    error_message = "Invalid Endpoint Type! Valid values are 'default', 'private', 'vpe', or 'link'"
+    condition     = contains(["default", "private", "vpe", "link"], var.cluster_config_endpoint_type)
+  }
+}
+
+variable "is_vpc_cluster" {
+  description = "Specify true if the target cluster for the monitoring agent is a VPC cluster, false if it is a classic cluster."
+  type        = bool
+  default     = true
 }
 
-variable "plan" {
+variable "wait_till" {
+  description = "To avoid long wait times when you run your Terraform code, you can specify the stage when you want Terraform to mark the cluster resource creation as completed. Depending on what stage you choose, the cluster creation might not be fully completed and continues to run in the background. However, your Terraform code can continue to run without waiting for the cluster to be fully created. Supported args are `MasterNodeReady`, `OneWorkerNodeReady`, `IngressReady` and `Normal`"
   type        = string
-  description = "The name of the plan type supported by service."
-  default     = "standard"
+  default     = "Normal"
+
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    error_message = "`wait_till` value must be one of `MasterNodeReady`, `OneWorkerNodeReady`, `IngressReady` or `Normal`."
+    condition = contains([
+      "MasterNodeReady",
+      "OneWorkerNodeReady",
+      "IngressReady",
+      "Normal"
+    ], var.wait_till)
   }
 }
 
-variable "resource_group_id" {
+variable "wait_till_timeout" {
+  description = "Timeout for wait_till in minutes."
+  type        = number
+  default     = 90
+}
+
+##############################################################################
+# Cloud Monitoring variables
+##############################################################################
+
+variable "cloud_monitoring_access_key" {
+  type        = string
+  description = "Access key used by the IBM Cloud Monitoring agent to communicate with the instance"
+  sensitive   = true
+  default     = null
+}
+
+variable "node_analyzer_enabled" {
```

Comment:
I don't think this should be exposed for this agent? We should check with Carlos. I think its only for workload protection - but lets see what he says

Reply:
confirmed, this should not be exposed